api/scpca_portal/models/job.py

```diff
@@ -224,46 +215,90 @@ def terminate_submitted(cls) -> List[Self]:
+        """
+        Updates the states of the synced jobs and their associated datasets.
+        """
+        cls.objects.bulk_update(
+            synced_jobs,
+            [
+                "state",
+                "submitted_at",
+                "succeeded_at",
+                "failed_at",
+                "failure_reason",
+                "terminated_at",
+            ],
+        )
+
+        Dataset.update_from_last_jobs([job.dataset for job in synced_jobs])
+
     @classmethod
     def bulk_sync_state(cls) -> bool:
         """
-        Sync all submitted job instances' states with the remote AWS Batch job statuses.
-        Update each job instance's state if it changes to COMPLETED, and update completed_at.
-        If the remote status is 'FAILED', update failure_reason if it hasn't been set already.
+        Syncs all submitted jobs' states with the remote AWS Batch job statuses.
+        Saves each job and its associated dataset if the state changes.
         """
-        if submitted_jobs := cls.objects.filter(state=JobStates.SUBMITTED.value):
-            if fetched_jobs := batch.get_jobs(submitted_jobs):
-                # Map the fetched AWS jobs for easy lookup by batch_job_id
-                aws_jobs = {job["jobId"]: job for job in fetched_jobs}
-
-                synced_jobs = []
-
-                for job in submitted_jobs:
-                    if aws_job := aws_jobs.get(job.batch_job_id):
-                        updated_job, updated = cls.update_job_state(job, aws_job)
-                        if updated:
-                            synced_jobs.append(updated_job)
-                        else:
-                            continue
-
-                if synced_jobs:
-                    cls.objects.bulk_update(
-                        synced_jobs,
-                        ["state", "failure_reason", "completed_at", "terminated_at"],
-                    )
+        submitted_jobs = cls.objects.filter(state=JobStates.SUBMITTED)
 
-                return True
+        if not submitted_jobs.exists():
+            return False
+
+        fetched_jobs = batch.get_jobs(submitted_jobs)
+
+        if not fetched_jobs:
+            return False
+
+        # Map the fetched AWS jobs for easy lookup by batch_job_id
+        aws_jobs = {job["jobId"]: job for job in fetched_jobs}
+
+        synced_jobs = []
+
+        for job in submitted_jobs:
+            if aws_job := aws_jobs.get(job.batch_job_id):
+                new_state, failure_reason = cls.get_job_state(aws_job)
+
+                if new_state != job.state:
+                    job.state = new_state
+                    job.failure_reason = failure_reason
+                    job.update_state_at()
+                    synced_jobs.append(job)
+                else:
+                    continue
+
+        if synced_jobs:
+            cls.bulk_update_state(synced_jobs)
+            return True
 
         return False
```

Comment:
I think this fits better with the pattern throughout the method. Also, it's typically good to follow the happy path.

```suggestion
        if not synced_jobs:
            return False

        cls.bulk_update_state(synced_jobs)
        return True
```

Reply:
I’ve updated the code to follow that pattern - thank you!

Reply:
I’ve updated the code to follow that happy path pattern in both `bulk_update_state`and `sync_state` - thank you!